File: src/70.cpp
class Solution {
public:
    int climbStairs(int n) {
        //干翻字节，老斐波那契了
        //老动态规划了，f(n)=f(n-1)+f(n-2)
        //做完这题回去吃饭了，然后准备干翻拼多多
        //三分钟，一次提交，时间击败-100%，空间击败-5.73%
        //题目保证n是一个正整数
        //边界情况
        if(n == 1){
            return 1;
        }
        else if(n == 2){
            return 2;
        }
        else{
            //一维动态规划，需要辅助数组
            //时间复杂度On，空间复杂度On
            vector<int> result(n);
            result[0] = 1;
            result[1] = 2;
            for(int i = 2; i < n; ++ i){
                result[i] = result[i - 1] + result[i - 2];
            }
            return result[n - 1];
        }
    }
};

File: src/70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        #干翻阿里
        #初级动态规划
        #n是正整数
        #三分钟，一次提交，时间击败-83.91%，空间击败-5.39%
        if n == 1:
            return 1
        elif n == 2:
            return 2
        else:
            #不递归，直接迭代
            result = [0 for _ in range(n)]
            result[0] = 1
            result[1] = 2
            for i in range(2, n):
                result[i] = result[i - 1] + result[i - 2]
            return result[-1]
